web/web.go

```diff
@@ -57,42 +58,48 @@ func (ws WebService) ServeForever() error {
 		http.Error(w, "", 404)
 	}))
 
-	http.Handle("/", prometheus.InstrumentHandler(
-		"/", ws.StatusHandler,
+	http.Handle(*urlPrefix, prometheus.InstrumentHandler(
+		*urlPrefix, ws.StatusHandler,
 	))
-	http.Handle("/alerts", prometheus.InstrumentHandler(
-		"/alerts", ws.AlertsHandler,
+	http.Handle(*urlPrefix + "alerts", prometheus.InstrumentHandler(
+		*urlPrefix + "alerts", ws.AlertsHandler,
 	))
-	http.Handle("/consoles/", prometheus.InstrumentHandler(
-		"/consoles/", http.StripPrefix("/consoles/", ws.ConsolesHandler),
+	http.Handle(*urlPrefix + "consoles/", prometheus.InstrumentHandler(
+		*urlPrefix + "/consoles/", http.StripPrefix("/consoles/", ws.ConsolesHandler),
 	))
-	http.Handle("/graph", prometheus.InstrumentHandler(
-		"/graph", http.HandlerFunc(graphHandler),
+	http.Handle(*urlPrefix + "graph", prometheus.InstrumentHandler(
+		*urlPrefix + "graph", http.HandlerFunc(graphHandler),
 	))
-	http.Handle("/heap", prometheus.InstrumentHandler(
-		"/heap", http.HandlerFunc(dumpHeap),
+	http.Handle(*urlPrefix + "heap", prometheus.InstrumentHandler(
+		*urlPrefix + "heap", http.HandlerFunc(dumpHeap),
 	))
 
-	ws.MetricsHandler.RegisterHandler()
-	http.Handle(*metricsPath, prometheus.Handler())
+	ws.MetricsHandler.RegisterHandler(urlPrefix)
+	http.Handle(*urlPrefix + *metricsPath, prometheus.Handler())
 	if *useLocalAssets {
-		http.Handle("/static/", prometheus.InstrumentHandler(
-			"/static/", http.StripPrefix("/static/", http.FileServer(http.Dir("web/static"))),
+		http.Handle(*urlPrefix + "static/", prometheus.InstrumentHandler(
+			*urlPrefix + "static/", http.StripPrefix(*urlPrefix + "static/", http.FileServer(http.Dir("web/static"))),
 		))
 	} else {
-		http.Handle("/static/", prometheus.InstrumentHandler(
-			"/static/", http.StripPrefix("/static/", new(blob.Handler)),
+		http.Handle(*urlPrefix + "static/", prometheus.InstrumentHandler(
+			*urlPrefix + "static/", http.StripPrefix(*urlPrefix + "static/", new(blob.Handler)),
 		))
 	}
 
 	if *userAssetsPath != "" {
-		http.Handle("/user/", prometheus.InstrumentHandler(
-			"/user/", http.StripPrefix("/user/", http.FileServer(http.Dir(*userAssetsPath))),
+		http.Handle(*urlPrefix + "user/", prometheus.InstrumentHandler(
+			*urlPrefix + "user/", http.StripPrefix(*urlPrefix + "user/", http.FileServer(http.Dir(*userAssetsPath))),
 		))
 	}
 
 	if *enableQuit {
-		http.Handle("/-/quit", http.HandlerFunc(ws.quitHandler))
+		http.Handle(*urlPrefix + "-/quit", http.HandlerFunc(ws.quitHandler))
+	}
+
+	if *urlPrefix != "/" {
```

Comment:
Ah, I guess this is a neat way of supporting both proxy configurations which drop the prefix as well as ones that don't?